pkg/sql/testdata/rename_index

```diff
@@ -74,3 +74,12 @@ SELECT * FROM users
 ----
 1 tom   cat
 2 jerry rat
+
+statement ok
+CREATE VIEW v AS SELECT name FROM users@{FORCE_INDEX=ufo}
+
+statement error cannot rename index "ufo" because it is depended on by view "test.v"
```

Comment:
That'd certainly be preferable behavior, but we've basically punted on query rewriting until we can have a proper semantic encoding of the view's query rather than the current text-based one - https://github.com/cockroachdb/cockroach/pull/9045#issuecomment-245769303

Unless you're suggesting that perhaps this particular case of rewriting is simple enough to handle?